noir-projects/noir-protocol-circuits/crates/public-kernel-lib/src/public_kernel_teardown.nr

```diff
@@ -3,13 +3,32 @@ use dep::types::abis::{
     kernel_circuit_public_inputs::{PublicKernelCircuitPublicInputs, PublicKernelCircuitPublicInputsBuilder},
     public_kernel_data::PublicKernelData, public_call_data::PublicCallData, gas_fees::GasFees
 };
+use dep::types::utils::arrays::array_to_bounded_vec;
 
 struct PublicKernelTeardownCircuitPrivateInputs {
     previous_kernel: PublicKernelData,
     public_call: PublicCallData,
 }
 
 impl PublicKernelTeardownCircuitPrivateInputs {
+    fn initialize_revert_code(self, circuit_outputs: &mut PublicKernelCircuitPublicInputsBuilder) {
+        // See https://docs.aztec.network/protocol-specs/gas-and-fees/kernel-tracking#handling-reverts
+        let previous_kernel_revert_code = self.previous_kernel.public_inputs.revert_code;
+        let public_call_revert_code = self.public_call.call_stack_item.public_inputs.revert_code;
+        if previous_kernel_revert_code.eq(0) & public_call_revert_code.eq(0) {
+            circuit_outputs.revert_code = 0;
+        } else if previous_kernel_revert_code.eq(1) & public_call_revert_code.eq(0) {
+            // Case where we carry forward a revert code from app logic
+            circuit_outputs.revert_code = 1;
+        } else if previous_kernel_revert_code.eq(0) & public_call_revert_code.eq(1) {
```

Comment:
Should we check the `public_call_revert_code` is non-zero rather than one? Or is it sure that AVM can only return 0 or 1 as revert codes?